src/Accordion.php

```diff
@@ -23,412 +16,487 @@
+     *
+     * @return string The HTML representation of the element.
+     *
+     * @psalm-param non-empty-string $idCollapse The ID of the collapse element.
      */
-    private function renderCollapse(array $item): Collapse
+    private function renderHeader(AccordionItem $accordionItem, string $idCollapse, bool $active): string
     {
-        $expand = $item['expand'] ?? false;
-        $options = $item['contentOptions'] ?? $this->contentOptions;
-        $toggleOptions = $item['toggleOptions'] ?? $this->toggleOptions;
-        $bodyOptions = $item['bodyOptions'] ?? $this->bodyOptions;
+        $headerAttributes = $this->headerAttributes;
+        $classesHeaderAttributes = $headerAttributes['class'] ?? null;
 
-        $toggleOptions['encode'] ??= $this->encodeLabels;
-        $bodyOptions['encode'] ??= $this->encodeTags;
+        unset($headerAttributes['class']);
 
-        Html::addCssClass($options, ['accordion-collapse']);
-        Html::addCssClass($toggleOptions, ['accordion-button']);
-        Html::addCssClass($bodyOptions, ['widget' => 'accordion-body']);
-
-        if (!$expand) {
-            Html::addCssClass($toggleOptions, ['collapsed']);
-        }
-
-        if ($this->autoCloseItems) {
-            $options['data-bs-parent'] = '#' . $this->getId();
+        if ($this->headerTag === '') {
+            throw new InvalidArgumentException('The "headerTag" property must be a non-empty string.');
         }
 
-        return Collapse::widget()
-            ->withToggleLabel($item['label'])
-            ->withToggleOptions($toggleOptions)
-            ->withOptions($options)
-            ->withContent($this->renderBody($item))
-            ->withBodyOptions($bodyOptions)
-            ->withCollapsed($expand)
-            ->withToggle(false);
+        return Html::tag($this->headerTag)
+            ->addAttributes($headerAttributes)
+            ->addClass(self::CLASS_HEADER, $classesHeaderAttributes)
+            ->addContent(
+                "\n",
+                $this->renderToggle($accordionItem->getHeader(), $idCollapse, $active),
+                "\n",
+            )
+            ->encode(false)
+            ->render();
     }
 
     /**
-     * Render collapse body
+     * Renders the items of the accordion component.
+     *
+     * @param string $idParent The ID of the parent element.
+     *
+     * @return string The HTML representation of the element.
+     *
+     * @psalm-param non-empty-string $idParent The ID of the parent element.
      */
-    private function renderBody(array $item): string
+    private function renderItems(string $idParent): string
     {
-        $items = '';
+        $items = [];
 
-        if ($this->isStringableObject($item['content'])) {
-            $content = [$item['content']];
-        } else {
-            $content = (array) $item['content'];
+        foreach ($this->items as $key => $item) {
+            $active = $key === 0 && !isset($this->cssClass['flush']);
```

Comment:
"flush" is appearance only: https://getbootstrap.com/docs/5.3/components/accordion/#flush

I think, it need to add "active" property to `AccordionItem` class.

Reply:
done.